addons/sale_timesheet/models/project.py

```diff
@@ -175,6 +177,17 @@ def _compute_sale_line_id(self):
             ], limit=1)
             project.sale_line_id = sol or project.sale_line_employee_ids.sale_line_id[:1]  # get the first SOL containing in the employee mappings if no sol found in the search
 
+    @api.depends('sale_line_id.product_uom_qty', 'sale_line_id.product_uom')
+    def _compute_allocated_hours(self):
+        porject_with_sol = self.filtered('sale_line_id')
+        (self - porject_with_sol).allocated_hours = 0
+        for rec in porject_with_sol:
+            rec.allocated_hours = rec.sale_line_id.product_uom._compute_quantity(
+                rec.sale_line_id.product_uom_qty,
+                self.env.company.timesheet_encode_uom_id,
+                raise_if_failure=False
+            )
```

Comment:
We could directly manage all cases in the for loop. 🙂 
Moreover, I prefer renaming `rec` into `project` since we know it is a project in the `self`.
Why don't we use the `timesheet_encode_uom_id` of company set into the project instead of using the context? 🤔 
```suggestion
        for project in self:
            if not project.sale_line_id:
                project.allocated_hours = 0
                continue
            project.allocated_hours = project.sale_line_id.product_uom._compute_quantity(
                project.sale_line_id.product_uom_qty,
                project.timesheet_encode_uom_id,
                raise_if_failure=False
            )
```